src/ir/ssa_names.h

```diff
@@ -0,0 +1,62 @@
+//-----------------------------------------------------------------------------
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//-----------------------------------------------------------------------------
+#ifndef SRC_IR_SSA_NAMES_H_
+#define SRC_IR_SSA_NAMES_H_
+
+#include "absl/container/flat_hash_map.h"
+
+namespace raksha::ir {
+
+class Block;
+class Operation;
+
+class SSANames {
```

Comment:
I think the C++ standard dictates that this should be `SsaNames`. See definition of "word" in the [general naming rules](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules) and the caps-case directive in [type naming rules](https://google.github.io/styleguide/cppguide.html#Type_Names).

Reply:
SSA abbreviates Static Single Assignment. So, I think this is OK.

Reply:
I know, but the general naming rules say that abbreviations count as words for the purpose of naming. You can override me if you want because I think it looks better as-is, but I think that's the standard.

Reply:
I went back and forth on this. In the end `SsaNames` seemed easier to parse. So, I went with it.